### PR TITLE
fix for some AMD Zen platforms

### PR DESCRIPTION
**Pull request by wkpark**
This is a quick hack to fix this case. (like as Asrok B450M-Pro4-F board)
- audio(AZAL) and video are in the same pci-bridge. Lilu does not detect it correctly.

Note:
> According to the code, Lilu will try to find an audio device. There are two types of audio:
> 1. audio with graphic (identified by v.video)
> 2. audio builtin onboard (assigned to variable `audioBuiltinAnalog`)
>
> from https://forums.unraid.net/topic/50191-video-guide-how-to-install-macos-mojave-or-high-sierra-as-a-vm/page/56/?tab=comments#comment-782474

Recent discussion:
- https://github.com/acidanthera/bugtracker/issues/296

gfxutil output with this fix.
```
% ./gfxutil -f HDEF
07:00.6 1022:15e3 /PCI0@0/GP17@8,1/HDEF@0,6 = PciRoot(0x0)/Pci(0x8,0x1)/Pci(0x0,0x6)
```

Screenshot with this patch.
image

### CODE REVIEW

File: Lilu/Sources/kern_devinfo.cpp [hunk 1]
```diff
@@ -256,7 +256,7 @@ void DeviceInfo::grabDevicesFromPciRoot(IORegistryEntry *pciRoot) {
 
 					pciiterator->release();
 
-					if (v.video) {
+					if (((v.audio && strcmp(v.audio->getName(), "AZAL")) || !v.audio) && v.video) { // check for some AMDZEN cases
```
Comment (vit9696) [inline Lilu/Sources/kern_devinfo.cpp:259]: Please add != 0 near strcmp and provide a more descriptive comment why this change in this needed right in the code prior to the if condition.

Like "On AMD Zen platforms… analog audio… video audio.… to avoid a conflict.…".

**Comment by hackmod** [comment]
thanks for your comment @vit9696

as I already mentioned above, this is a quick hack and a bug report (since there is no issue tracker)
so code might be somewhat hacky.

and as you already commit/comment 2-years ago at
https://github.com/acidanthera/Lilu/commit/5f147cf7cda622c6b7516a67a8c4c374cf4a4923#diff-ac14a6d9b9c473a9ae438f5cc93cc218e9e6574e8143389bcac5e2a066984d84R280
(see also https://github.com/acidanthera/Lilu/pull/49)

> On modern AMD platforms built-in audio devices sit on a PCI bridge just any other device

for my case, `AZAL`(analog audio. `v.audio`) and `VGA@0` (internal GPU, `v.video`) reside on the same bridge (please see the screenshot)

and recent changes https://github.com/acidanthera/Lilu/commit/9d50e555ca4b2437b537b9d40e3543608c84c2fc break this behavior (I guess)

I'm not sure this is a correct or optimized fix for all AMD Zen but this is a working fix for me.

File: Lilu/Sources/kern_devinfo.cpp (hunk 1 above)
Comment (hackmod) [reply Lilu/Sources/kern_devinfo.cpp:259]: thanks for your comment, will fix asap

Comment (wkpark) [reply Lilu/Sources/kern_devinfo.cpp:259]: thanks for your comment, will fix asap

**Comment by wkpark** [comment]
thanks for your comment @vit9696

as I already mentioned above, this is a quick hack and a bug report (since there is no issue tracker)
so code might be somewhat hacky.

and as you already commit/comment 2-years ago at
5f147cf#diff-ac14a6d9b9c473a9ae438f5cc93cc218e9e6574e8143389bcac5e2a066984d84R280
(see also #49)

On modern AMD platforms built-in audio devices sit on a PCI bridge just any other device

for my case, AZAL(analog audio. v.audio) and VGA@0 (not working internal GPU, v.video) reside on the same bridge (please see the screenshot)

and recent changes 9d50e55 break this behavior (I guess)

I'm not sure this is a correct or optimized fix for all AMD Zen but this is a working fix for me.

**Comment by vit9696** [comment]
We check for some device names above, so while hardcoding AZAL might not be great, it is a fair fix for the time being.
